streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableKTableAbstractJoin.java

```diff
@@ -39,9 +39,15 @@
     }
 
     @Override
-    public final void enableSendingOldValues() {
-        table1.enableSendingOldValues();
-        table2.enableSendingOldValues();
+    public final boolean enableSendingOldValues(final boolean onlyIfMaterialized) {
+        if (!table1.enableSendingOldValues(onlyIfMaterialized)) {
```

Comment:
Instead of the check, should we pass in `false` to enforce a materialization if necessary?

Reply:
My view, as outlined above, is to stick with the pattern that's here.

Reply:
My comment was about your pattern... (if we flip the logic, we would need to pass in `true` to force a materialization). My point is, shouldn't we pass in a constant? For KTableKTableAbstractJoin we always want that the upstream is sending us old values.

Reply:
Feels a little off to me to ignore what the caller is asking.  If they pass `true` for `onlyIfMaterialized`, then we shouldn't silently ignore them IMHO.

Reply:
Anyway, done...